src/Javis.jl

```diff
@@ -270,121 +269,120 @@ function javis(
+        CURRENT_OBJECT[1] = object
+        if frame in get_frames(object)
+            # check if the object should be part of the global layer (i.e BackgroundObject)
             # or in its own layer (default)
-            in_global_layer = get(action.opts, :in_global_layer, false)::Bool
+            in_global_layer = get(object.opts, :in_global_layer, false)::Bool
             if !in_global_layer
                 @layer begin
-                    perform_action(action, video, frame, origin_matrix)
+                    draw_object(object, video, frame, origin_matrix)
                 end
             else
-                perform_action(action, video, frame, origin_matrix)
+                draw_object(object, video, frame, origin_matrix)
                 # update origin_matrix as it's inside the global layer
                 origin_matrix = cairotojuliamatrix(getmatrix())
             end
         end
-        # if action is in global layer this changes the background settings
-        update_background_settings!(background_settings, action)
+        # if object is in global layer this changes the background settings
+        update_background_settings!(background_settings, object)
     end
     img = image_as_matrix()
     finish()
     return img
 end
 
 """
-    perform_action(action, video, frame, origin_matrix)
+    draw_object(object, video, frame, origin_matrix)
 
-Is called inside the `javis` and does everything handled for an `AbstractAction`.
+Is called inside the `javis` and does everything handled for an `AbstractObject`.
 It is a 4-step process:
-- compute the transformation for this action (translation, rotation, scale)
-- do the transformation
-- call the action function
-- save the result of the action if wanted inside `video.defs`
+- translate to the start position
+- call the relevant actions
+- call the object function
+- save the result of the object if wanted inside `video.defs`
 """
-function perform_action(action, video, frame, origin_matrix)
-    # first compute and perform the global transformations of this action
-    compute_transition!(action, video, frame)
-    perform_transformation(action)
-
-    # relative frame number for subactions
-    rel_frame = frame - first(get_frames(action)) + 1
-    # call currently active subactions and their transformations
-    for subaction in action.subactions
-        if rel_frame in get_frames(subaction)
-            subaction.func(video, action, subaction, rel_frame)
-            compute_transition!(subaction, video, rel_frame)
-            perform_transformation(subaction)
-        elseif rel_frame > last(get_frames(subaction))
-            # call the subaction on the last frame i.e. disappeared things stay disappeared
-            subaction.func(video, action, subaction, last(get_frames(subaction)))
+function draw_object(object, video, frame, origin_matrix)
+    # first compute and perform the global transformations of this object
+    translate(object.start_pos)
```

Comment:
Is the preceding comment line still correct? All you are doing here it appears is translating the object to its start position on the frame.

Reply:
The comment line should move down. Will fix